### PR TITLE
Refactor hardware-capabilities-based tactic assignment

### LINKED ISSUE #753: Refactor robot capabilities-based assignment to use a set of RobotCapability
### Description of the task

Refactor robot capabilities-based assignment to use a set of RobotCapability

### Outline the task in more detail

Let's get rid of this `std::bitset` based `RobotCapabilityFlags` class and just use a set of a `RobotCapability` enum type

### Acceptance criteria

- [ ] the task is done

### PR DESCRIPTION
**Pull request by jordanzliu**
### Description

- removed `RobotCapabilityFlags` class 
- changed usages of `RobotCapabilityFlags` to a set of an enum `RobotCapabilities::Capability` 
- added `Capability::Move` for robots that are able to move properly
- added `Move` as a default requirement for all tactics in the tactic constructor
- implemented subset and superset operators for std::set in `robot_capabilities.h` 

### Testing Done

Existing and new unit tests

### Resolved Issues

Resolves #753 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — DISMISSED** [review]
LGTM pending CI

**Review by EvanMorcom — DISMISSED** [review]
👍 
LGTM
Just need CI

**Review by henryymliu — DISMISSED** [review]
LGTM, but have fun with merge conflicts

File: src/thunderbots/software/test/ai/world/robot_capabilities.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "ai/world/robot_capabilities.h"
```
Comment (jonathanlew) [inline src/thunderbots/software/test/ai/world/robot_capabilities.cpp]: Can we move this to src/thunderbots/software/ai/world/robot_capabilities_test.cpp? We've changed the folder structure so we don't have a test folder anymore

**Review by jonathanlew — CHANGES_REQUESTED** [review]
LGTM! Just a quick folder structure change.

**Comment by MathewMacDougall** [comment]
LGTM pending Jon's comment. Still have a test failure. @jordanzliu can you still not reproduce it locally?

**Comment by MathewMacDougall** [comment]
Still seeing the same test failure :thinking:

**Comment by jordanzliu** [comment]
I've literally copypasted the command out of travis.yml and it still doesn't fail on my machine. Gonna try spinning up a Ubuntu docker container to test it, but i doubt that'll do anything

File: src/thunderbots/software/test/ai/world/robot_capabilities.cpp (hunk 1 above)
Comment (jordanzliu) [reply src/thunderbots/software/test/ai/world/robot_capabilities.cpp]: fixed

**Review by MathewMacDougall — APPROVED** [review]
Oh bless CI is passing

**Review by jonathanlew — APPROVED** [review]
:+1:

**Comment by MathewMacDougall** [comment]
:shipit: :shipit: :shipit:

**Pull request merged**